crates/provider/src/heart.rs

```diff
@@ -448,38 +463,119 @@ impl<S> Heartbeat<S> {
         }
     }
 
+    /// Reap transactions overridden by the reorg.
+    fn reap_reorg(&mut self, new_height: u64) {
```

Comment:
what does this do?
needs some additional docs, not obvious what this does?

looks like this drops the channel? which would mean on reorg, the listeners is unable to obtain the receipt because the watcher dropped?

Reply:
> what does this do?

It erases all the subscriptions for the blocks that were affected by the reorg. Basically, it works the same as `reap_timeouts` above.

> looks like this drops the channel?

It does, similarly to how it does on timeout. If the block with your tx was reorganized, you are not guaranteed to receive a receipt for it, so IMHO it makes sense to drop it.

A dedicated error type would be nice (even if just for timeout), but it's a change in public interfaces, and not sure if it belongs to this PR (I can submit it in a separate PR though).

Reply:
Added more information to the method doc-comment.

Reply:
> so IMHO it makes sense to drop it.

I don't think this makes sense, because there's a high chance this tx will also be included in the new chain.

Reply:
I see, so would putting it back to `unconfirmed` make sense then?